torch/_dynamo/variables/functions.py

```diff
@@ -639,3 +639,70 @@ def as_python_constant(self):
                 *[arg.as_python_constant for arg in self.args],
                 **{k: v.as_python_constant() for k, v in self.keywords.items()},
             )
+
+
+class TritonKernelVariable(VariableTracker):
+    def __init__(self, kernel, grid, **kwargs):
+        super().__init__(**kwargs)
+        self.kernel = kernel
+        self.grid = grid
+
+    def call_function(
+        self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
+    ) -> "VariableTracker":
+        def call_kernel(grid, *args, **kwargs):
+            self.kernel.run(*args, grid=grid, **kwargs)
+
+        from .dicts import ConstDictVariable
+        from .functions import NestedUserFunctionVariable, UserFunctionVariable
+        from .lists import BaseListVariable
+
+        grid = self.grid
+
+        # If the grid is a function, then lets execute it and convert it to
+        # a list
+        if isinstance(grid, (NestedUserFunctionVariable, UserFunctionVariable)):
```

Comment:
These two function variables and list variables (as per below) are supported. Do I need to support any other type? Triton's docs only ever use lambda and tuple.

Reply:
I think this is fine, we can add more later if needed.